pkg/webhook/certs.go

```diff
@@ -151,15 +162,15 @@ func (cr *certRotator) refreshCerts(refreshCA bool, secret *corev1.Secret) (bool
 	}
 	writeFn := func() (bool, error) {
 		if err := cr.writeSecret(cert, key, caArtifacts, secret); err != nil {
-			return false, err
```

Comment:
I think it was a bug to return the `err` here, since it will abort the backoff loop. I changed it to `nil` if the secret was not able to be updated, so it will backoff and retry.

Reply:
Good call.

I think the retry with exponential backoff needs to be around refreshCertsIfNeeded, as the value of `refreshCA` could change depending on the value of the re-read secret.

Reply:
Ah right! I added the backoff to `refreshCertsIfNeeded` PTAL.